feat: Token Security Alert

limichange opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced comprehensive token security alerts on the Market Detail page, offering detailed analysis including honeypot detection, tax rates, and ownership checks.
  - Added an interactive dialog presenting an overview and detailed breakdown of security checks with clear visual indicators for warnings and safe status.
  - Security alerts are now visible within the token details header for supported tokens.
- **Bug Fixes**
  - Improved layout spacing on the Market Detail page for better readability.
- **Chores**
  - Expanded the spell checker’s skip word list with additional terms.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

limichange committed:
feat: Add Token Security Alert component to MarketDetailV2

- Introduced TokenSecurityAlert component to display security alerts for tokens.
- Integrated TokenSecurityAlert into TokenDetailHeader, showing alerts based on token address and network ID.
- Implemented fetching of token security data in ServiceMarketV2 with a new method.
- Updated types to include IMarketTokenSecurity for enhanced token security information.
- Enhanced layout in MarketDetailV2 for better organization of components.

limichange committed:
feat: Refactor TokenSecurityAlert to use custom hook for data fetching

- Introduced useTokenSecurity hook to encapsulate the logic for fetching token security data.
- Updated TokenSecurityAlert component to utilize the new hook, improving separation of concerns.
- Modified TokenSecurityAlertDialogContent to accept security data, error, and loading state as props, enhancing its reusability.
- Improved loading and error handling in the TokenSecurityAlert component.

limichange committed:
feat: Enhance useTokenSecurity hook with security status and warning count

- Added securityStatus and warningCount to the useTokenSecurity hook to provide detailed security insights for tokens.
- Implemented analyzeSecurityData function to evaluate token security data and determine the security status based on various warning indicators.
- Updated error handling to reset security status and warning count when fetching fails or no token address is provided.

limichange committed:
feat: Update TokenSecurityAlert component to display security status and warning count

- Enhanced TokenSecurityAlert to utilize securityStatus and warningCount from the useTokenSecurity hook.
- Refactored the component layout to use ButtonFrame and XStack for improved UI presentation.
- Updated icon color based on security status to provide clearer visual feedback.

limichange committed:
refactor: Remove hasSelfdestruct from useTokenSecurity hook

- Eliminated the hasSelfdestruct property from the useTokenSecurity hook to streamline security data evaluation.
- This change enhances the clarity and focus of the security checks performed within the hook.

limichange committed:
feat: Enhance TokenSecurityAlert with formatted security data display

- Added formatSecurityData function to transform token security data into a structured format for better presentation.
- Updated TokenSecurityAlertDialogContent to utilize formatted security data, improving the clarity of security alerts.
- Enhanced loading and error states with improved UI elements for a better user experience.

limichange committed:
feat: Add overview section to TokenSecurityAlertDialog for enhanced security insights

- Introduced TokenSecurityAlertDialogContentOverview component to display warning count and total checks.
- Updated TokenSecurityAlertDialogContent to integrate the new overview section, improving the presentation of security data.
- Adjusted icon usage and colors for better visual feedback on security status.

limichange committed:
feat: Refactor TokenSecurityAlertDialog to improve component structure and clarity

- Added TokenSecurityAlertDialogContentItem and TokenSecurityAlertDialogContentIcon components to streamline the rendering of security alert items.
- Updated TokenSecurityAlertDialogContent to utilize the new components, enhancing readability and maintainability of the code.
- Included 'isMintable' in the spell checker skip words list for improved spell checking accuracy.

limichange committed:
refactor: Update spell checker skip words and streamline security data analysis

- Removed unnecessary entries from the spell checker skip words list for improved accuracy.
- Moved the analyzeSecurityData function to a more appropriate location, enhancing code organization and readability.
- Added missing warning indicators to the analyzeSecurityData function for comprehensive security checks.

limichange committed:
Merge branch 'x' into market-detail-page-6

originalix commented on packages/kit/src/views/Market/MarketDetailV2/components/TokenSecurityAlert/TokenSecurityAlertDialogContent.tsx:
```diff
@@ -0,0 +1,91 @@
+import type { FC } from 'react';
```
ai 写的嘛，最好统一都不用 FC 的写法了

limichange replied:
我把 AI 规则更新一下，我看项目里其他很多地方也是 FC 写法，估计是读到了这些地方。

limichange committed:
refactor: Simplify functional component definitions in TokenSecurityAlert

- Converted TokenSecurityAlert, TokenSecurityAlertDialogContent, TokenSecurityAlertDialogContentIcon, and TokenSecurityAlertDialogContentItem components from arrow functions to standard function declarations for improved readability and consistency.
- Removed unnecessary type imports to streamline the codebase.

originalix committed:
Merge branch 'x' into market-detail-page-6

limichange committed:
Merge branch 'x' into market-detail-page-6

limichange committed:
Merge branch 'x' into market-detail-page-6

limichange committed:
Merge branch 'x' into market-detail-page-6

limichange committed:
Merge branch 'x' into market-detail-page-6
